apps/services/regulations-admin-backend/src/app/modules/draft_regulation/draft_regulation.service.ts

```diff
@@ -0,0 +1,404 @@
+    create: CreateDraftRegulationDto,
+    user?: User,
+  ): Promise<DraftRegulationModel> {
+    this.logger.debug('Creating a new DraftRegulation')
+
+    const createData: Partial<DraftRegulationModel> = {
+      drafting_status: 'draft',
+      title: '',
+      text: '',
+      drafting_notes: '',
+      type: create.type,
+      authors: [user?.nationalId as Kennitala],
+    }
+
+    return this.draftRegulationModel.create(createData)
+  }
+
+  async update(
+    id: string,
+    update: UpdateDraftRegulationDto,
+    user?: User,
+  ): Promise<{
+    numberOfAffectedRows: number
+    updatedDraftRegulation: DraftRegulationModel
+  }> {
+    this.logger.debug(`Updating DraftRegulation ${id}`)
+
+    const nationalId = user?.nationalId as Kennitala
+    const draftRegulation = await this.findDraftById(id)
+
+    if (draftRegulation && !draftRegulation.authors.includes(nationalId)) {
+      update.authors = [...draftRegulation.authors, nationalId]
+    }
+
+    const updateData: Partial<DraftRegulationModel> = {
+      name: update.name,
+      title: update.title,
+      text: update.text,
+      appendixes: update.appendixes,
+      comments: update.comments,
+      ministry: update.ministry,
+      drafting_notes: update.draftingNotes,
+      ideal_publish_date: update.idealPublishDate,
+      law_chapters: update.lawChapters,
+      signature_date: update.signatureDate,
+      signature_text: update.signatureText,
+      effective_date: update.effectiveDate,
+      type: update.type,
+      drafting_status: update.draftingStatus,
+      signed_document_url: update.signedDocumentUrl,
+      authors: update.authors,
+      fast_track: update.fastTrack,
+    }
+
+    const [
+      numberOfAffectedRows,
+      [updatedDraftRegulation],
+    ] = await this.draftRegulationModel.update(updateData, {
+      where: { id },
+      returning: true,
+    })
+
+    return { numberOfAffectedRows, updatedDraftRegulation }
+  }
+
+  async delete(id: string): Promise<number> {
+    this.logger.debug(`Deleting DraftRegulation ${id}`)
+
+    // destroy all draft regulation impacts
+    await this.draftRegulationCancelService.deleteRegulationDraftCancels(id)
+    await this.draftRegulationChangeService.deleteRegulationDraftChanges(id)
+
+    return this.draftRegulationModel.destroy({
+      where: {
+        id,
+      },
+    })
+  }
+
+  async getAuthorsInfo(nationalIds: Kennitala[]): Promise<Author[]> {
+    const authors: Author[] = []
+    for await (const nationalId of nationalIds) {
+      if (kennitala.isCompany(nationalId)) {
+        continue
+      }
+      let author = await this.draftAuthorService.get(nationalId)
+
+      if (!author) {
+        try {
+          const person = await this.nationalRegistryApi.getUser(nationalId)
+          if (person) {
+            author = {
+              name: person.Fulltnafn,
+              authorId: nationalId,
+            }
+            await this.draftAuthorService.create(author)
+          }
```

Comment:
We're trying to get rid of all soffia dependencies. Can we update this to use the v2 service?